concordium-node/src/configuration.rs

```diff
@@ -881,10 +881,10 @@ pub struct AppPreferences {
 
 impl AppPreferences {
     /// Creates an `AppPreferences` object.
-    pub fn new(override_conf: PathBuf, override_data: PathBuf) -> Self {
+    pub fn new(override_conf: PathBuf, override_data: PathBuf) -> anyhow::Result<Self> {
         let file_path = Self::calculate_config_file_path(&override_conf, APP_PREFERENCES_MAIN);
-        let mut new_prefs = match OpenOptions::new().read(true).write(true).open(&file_path) {
-            Ok(file) => {
+        let mut new_prefs =
+            if let Ok(file) = OpenOptions::new().read(true).write(true).open(&file_path) {
```

Comment:
It looks like in this branch you only need read access to the file, so unless I'm missing something you could just do
```rust
if let Ok(file) = File::open(&file_path) {
...
```